content/blog/2021/04/2021-04-20-tekton-plugin.adoc

```diff
@@ -0,0 +1,178 @@
+
+The link:https://jenkins-x.io/[Jenkins X project] automates your CI/CD on kubernetes to help you link:https://jenkins-x.io/v3/devops/accelerate/[accelerate]:
+
+* link:https://jenkins-x.io/v3/develop/create-project/[Automated CI/CD pipelines] lets you focus on your actually application code while Jenkins X automatically creates battle tested https://github.com/tektoncd/pipeline[Tekton] CI/CD pipelines for your project which are link:https://jenkins-x.io/blog/2021/02/25/gitops-pipelines/[managed via GitOps] so that its super easy to keep your pipelines up to date across your repositories or to upgrade or link:https://jenkins-x.io/v3/develop/pipelines/catalog/#overriding-a-pipeline-step-locally[override pipelines or steps] for specific repositories.
+* link:https://jenkins-x.io/v3/develop/environments/promotion[Automatic promotion of versioned artifacts] via link:https://jenkins-x.io/v3/devops/patterns/gitops/[GitOps] through your link:https://jenkins-x.io/v3/develop/environments/[Environments] such as `Staging`, `Pre-production` and `Production` whether they are running in the same kubernetes cluster or you are using link:https://jenkins-x.io/v3/admin/guides/multi-cluster/[multiple clusters for your environments]
+* link:https://jenkins-x.io/v3/develop/environments/preview/[Preview Environments] lets you propose code changes via Pull Requests and have a Preview Environment automatically created, running your code in kubernetes to get fast feedback from your team before agreeing to merge changes to the main branch
+* link:https://jenkins-x.io/v3/develop/developing/#using-chatops[ChatOps] comment on Pull Requests to give feedback, approve/hold changes, trigger optional pipelines for additional testing and other link:https://jenkins-x.io/v3/develop/reference/chatops/[ChatOps commands]
+
+All of the above is implemented in reusable Tekton pipelines.
+
+=== Reusing Jenkins X Pipelines
+
+So how can we reuse automated CI/CD pipelines from link:https://jenkins-x.io/[Jenkins X project] from Jenkins?
+
+Make sure you have the link:https://github.com/jenkinsci/tekton-client-plugin[tekton plugin for jenkins] installed in your Jenkins server.
+
+==== Using a working template
+
+If you want to start with a working example then
+
+* link:https://github.com/jstrachan/node-example/generate[Create A Git Repository From This Template]
+
+* add a new `Frestyle project` to your Jenkins server
+* enable the `Git` source code management for your new github.com repository
+* click `Add build Step` (near the bottom of the page) and then select `Tekton : Create Resource (Raw)`
+* make sure that `FILE` is selected for the input and enter the name `.lighthouse/jenkins-x/release.yaml` for the file name
+* if you are using a Jenkins X cluster enter `jx` for the namespace
+* ensure that `Enable Tekton Catalog` is checked
+* now save the pipeline - it should look something like this:
+
+
+image:/images/tekton-plugin/example.png["Jenkins Console"]
```

Comment:
Likely the image could be also compressed

Reply:
I've shrunk it 50% for now - that ok?

Reply:
Yes, thanks! Soon we should have a stable GitHub action for that